### ISSUE TITLE
Legend tag rendered for form inputs header is missing inner span tag

### ISSUE THREAD

**Issue by denisahearn** [activeadmin/activeadmin#8466]
We recently upgraded our Rails app from `activeadmin 2.14.0` to `activeadmin 3.2.4`, and noticed that the CSS styling of the form inputs section broke.

**activeadmin 2.14.0**
![activeadmin-2 14 0](https://github.com/user-attachments/assets/30391f33-0ab2-46a8-9083-2373202de53e)

**activeadmin 3.2.4**
![activeadmin-3 2 4](https://github.com/user-attachments/assets/79f5aec5-a919-4528-9a59-44177296e4d6)

Digging into this, I believe this defect was introduced by this change: https://github.com/activeadmin/activeadmin/commit/55d88a6f230979185a975ba09c3dfe6b8a8d9ce9#diff-e1a3586b0ace3c17fec674be8fd38d0c2937ac81cc961006bf302e6de4b78fe6R130

With `2.14.0`, the form inputs header is rendered as this HTML:

```
<fieldset class="inputs">
  <legend>
    <span>User Information</span>
  </legend>
  ...
</fieldset>
```

With `3.2.4`, the form inputs header is rendered as this HTML:

```
<fieldset class="inputs">
  <legend class="fieldset-title">User Information</legend>
  ...
</fieldset>
```

I am using this monkey patch to work around this issue for now:

```
module ActiveAdmin
  module Views
    class SemanticInputsProxy < FormtasticProxy
      def build(_form_builder, *args, &block)
        html_options = args.extract_options!
        html_options[:class] ||= 'inputs'
        legend = args.shift if args.first.is_a?(::String)
        legend = html_options.delete(:name) if html_options.key?(:name)
        # =============================
        # Add back the missing span tag
        legend_tag = legend ? helpers.tag.legend(helpers.tag.span(legend), class: 'fieldset-title') : ''
        # =============================
        fieldset_attrs = tag_attributes html_options
        @opening_tag = "<fieldset #{fieldset_attrs}>#{legend_tag}<ol>"
        @closing_tag = '</ol></fieldset>'
        super(*(args << html_options), &block)
      end
    end
  end
end
```

### Expected behavior

A form inputs header should be styled the same that it was in `activeadmin 2.14.0`

### Actual behavior

A form inputs header is missing its styling

### How to reproduce

Using `rails 7.2.1` and `activeadmin 3.2.4`, create an ActiveAdmin form like so:

```
form(data: { address_form: true }) do |f|
  inputs 'User Information' do
    f.input :first_name
    f.input :last_name
  end

  f.submit 'Save`
end
```

Note how the form inputs header is rendered/styled incorrectly.

**Comment by javierjulio** [comment]
@denisahearn thank you. We will accept a PR to fix this. It should be a straightforward change.

**Comment by denisahearn** [comment]
@javierjulio I stopped short of opening a PR because I don't want to assume that the bug should be fixed by adding back a `span` tag, vs. changing the CSS styling to style the `legend` tag directly. 

If adding back the missing `span` tag is the desired fix, then sure, I'll open a PR for that.

Also, it looks like this bug was introduced in `3.x` by back-porting a change that I assume originated in the `4.0` work. Is that true, and if so, then do I need to open two PRs for this?

**Comment by mgrunberg** [comment]
@denisahearn thanks for opening the issue. You are right, removing the `span` was an unwanted side-effect of #https://github.com/activeadmin/activeadmin/pull/8446. The `span` [is not present in v4](https://github.com/activeadmin/activeadmin/pull/8439) but should be in v3.

Would you like to open the PR adding it back? I can cut a release after that.

Note: I guess we can add `span` to [this selector](https://github.com/activeadmin/activeadmin/pull/8446/files#diff-fc9fc54227bb22cc4a95871a1d468220d399e31ac3d8dfceb325efe83caf3cc8R85) to make sure this doesn't happen again in v3

**Comment by javierjulio** [comment]
@denisahearn yes, as @mgrunberg stated, adding the `span` back is what we need to fix the issue. Please open a PR against the 3-0-stable branch. This will not affect v4 since we removed the span tag usage due to the styling changes.

**Comment by denisahearn** [comment]
@javierjulio @mgrunberg Ok, I'll open a PR against the 3-0-stable branch. Thanks for clarifying.

**Comment by denisahearn** [comment]
https://github.com/activeadmin/activeadmin/pull/8467

**Comment by mgrunberg** [comment]
I'm closing the issue because it's fixed. The release will happen soon.

**Comment by mgrunberg** [comment]
For future reference, the fix was released in 3.2.5
